server/server.ts

```diff
@@ -133,6 +161,300 @@ app.post('/updateCourse', async (req, res) => {
+ * Postcondition: returns an an array of the desired major(s) with required fields 
+ */
+app.get('/getMajors', async (req, res) => {
+  const majorTitle = req.query.title;
+  const localMajors: Major[] = [];
+
+  if (majorTitle === null) {
+    const majorDocs = await majors.get();
+
+    for (const docRef of majorDocs.docs) {
+      const majorC: majorContent = docRef.data() as majorContent;
+      const major: Major = {
+        content: majorC,
+        title: docRef.id,
+      };
+      localMajors.push(major);
+    }
+
+    res.send({ success: true, data: localMajors });
+  } else {
+    const majorDocRef = await majors.doc(majorTitle.toString()).get();
+    const majorC: majorContent = majorDocRef.data() as majorContent;
+    if (majorC === null) {
+      res.send({ success: false, message: 'Major not found.' });
+    } else {
+      const major: Major = {
+        content: majorC,
+        title: majorDocRef.id,
+      };
+      localMajors.push(major);
+      res.send({ success: true, data: localMajors });
+    }
+  }
+});
+
+/**
+ * creates a new major object in firestore using client provIded fields
+ * Precondition: Client must provIde all required fields for the major,
+ *               must not be a duplicate of an existing major
+ * Postcondition: One new major will be created and stored in firestore
+ */
+app.post('/createMajor', async (req, res) => {
+  const major: Major = req.body;
+
+  if (
+    major.title === null ||
+    major.content.academicLevel === null ||
+    major.content.departmentPage === null ||
+    major.content.designAreas === null ||
+    major.content.reasons === null ||
+    major.content.school === null ||
+    major.content.type === null
+  ) {
+    res.send({ success: false, message: 'one or more fields is missing' });
+  } else {
+    const newMajor = majors.doc(major.title);
+    newMajor.set(major.content);
+    res.send({ success: true, data: major });
+  }
+});
+
+/**
+ * querying the database for the major with the major title
+ * Precondition: request body must have a major title
+ * Postcondition: only a singular major will be deleted
+ */
+app.delete('/deleteMajor', async (req, res) => {
+  const { title } = req.body;
+
+  if (title === null) {
+    res.send({ success: false, message: 'One or more fields is missing' });
+  } else {
+    majors.doc(title).delete();
+    res.send({ success: true });
+  }
+});
+
+/**
+ * updates the specified field of a major with specified content
+ * Precondition: request body must have a major title, field, and new content
+ * Postcondition: the specified data of a singular major will be updated
+ */
+app.post('/updateMajor', async (req, res) => {
+  const { field } = req.body;
+  const { title } = req.body;
+  const { content } = req.body;
+
+  if (content === null || field === null || title === null) {
+    res.send('One or more fields is missing.');
+  } else {
+    majors.doc(title).update({ field: content });
+    res.send({ success: true, data: [] });
```

Comment:
If data is just going to be an empty array, it might be better to just not include it when sending a response back.

Reply:
Thanks for the comment, I agree with that change and have implemented it!